pkg/driver/controller.go

```diff
@@ -243,6 +243,42 @@ func (cs *controller) init() error {
 	return nil
 }
 
+func validateRestoreRequestFromSnapshot(lvmSnapCr *lvmapi.LVMSnapshot,
+	req *csi.CreateVolumeRequest,
+	params *VolumeParams,
+) error {
+	// verify that the snapshot is of thin type
+	// only thin snapshot can be used to restore a volume
+	// also verify that the new volume to be created from snapshot
+	// is also thin provisioned
+	if !lvmSnapCr.Spec.Thin {
+		return status.Errorf(codes.Internal, "snapshot %s is not of thin type", lvmSnapCr.Name)
+	} else if params.ThinProvision != lvm.YES {
+		return status.Errorf(codes.Internal, "only thin restores supported today.")
```

Comment:
Shouldn't these be `INVALID_ARGUMENT` ? See the [spec](https://github.com/container-storage-interface/spec/blob/e6fc13ea4d529db12e211ef79c924ee3186c39d5/spec.md?plain=1#L1262)
 
```suggestion
		return status.Errorf(codes.InvalidArgument, "only thin restores supported today")
```

Reply:
`InvalidArgument` if Volume is a content source? We return `Unimplemented` now.

Reply:
Yes that's right. 
`Unimplemented` is likely not correct, as that's for `RPC handler` not implemented.

Reply:
changed it to `InvalidArgument `